lib/nexpose/site.rb

```diff
@@ -241,16 +216,19 @@ def remove_asset(asset)
       begin
         # If the asset registers as a valid IP, remove as IP.
         IPAddr.new(asset)
-        remove_ip(asset)
+        @assets = assets.reject { |ip| ip == IPRange.new(asset) }
       rescue ArgumentError => e
```

Comment:
Instead of reducing ArgumentError and using `IPAddr.new` why not use the following:

```ruby
# Use Array#reject! or Array#delete_if (equivalent) to modify the existing array.
assets.reject! { |existing_asset| existing_asset == HostOrIP.convert(asset) }
```